backend/dal/Services/Concrete/ProjectReportService.cs

```diff
@@ -0,0 +1,258 @@
+                this.Context.ProjectSnapshots.AddRange(GenerateSnapshots((DateTime)report.To));
+            }
+
+            this.Context.SaveChanges();
+            this.Context.CommitTransaction();
+
+            return Get(originalReport.Id);
+        }
+
+        /// <summary>
+        /// Generate an updated list of snapshots based on current project data.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <exception cref="NotAuthorizedException">User does not have permission to get refreshed snapshots.</exception>
+        /// <returns></returns>
+        public IEnumerable<ProjectSnapshot> Refresh(int reportId)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.AdminProjects);
+            var report = Get(reportId);
+            var currentSnapshots = this.Context.ProjectSnapshots
+                .Include(s => s.Project).ThenInclude(p => p.Agency)
+                .Include(s => s.Project).ThenInclude(p => p.Status)
+                .Where(s => s.SnapshotOn == report.To)
+                .ToArray();
+
+            var snapshots = GenerateSnapshots(DateTime.UtcNow).Concat(currentSnapshots);
+            foreach (ProjectSnapshot snapshot in snapshots)
+            {
+                var previousSnapshot = snapshots.Where(s => s.ProjectId == snapshot.ProjectId && s.Id != snapshot.Id).FirstOrDefault();
+                snapshot.BaselineIntegrity = (snapshot?.NetProceeds ?? 0) - (previousSnapshot?.NetProceeds ?? 0);
+            }
+
+            return snapshots;
+        }
+
+        /// <summary>
+        /// Remove the specified report from the datasource.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <exception cref="ArgumentNullException">Argument 'report' is required.</exception>
+        /// <exception cref="RowVersionMissingException">Project report rowversion is required.</exception>
+        /// <exception cref="KeyNotFoundException">Project report does not exist.</exception>
+        /// <exception cref="NotAuthorizedException">User does not have permission to delete report.</exception>
+        /// <returns></returns>
+        public void Remove(ProjectReport report)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.AdminProjects);
+
+            var originalReport = this.Context.ProjectReports
+                .SingleOrDefault(p => p.Id == report.Id) ?? throw new KeyNotFoundException();
+            if (originalReport.IsFinal) throw new InvalidOperationException($"Unable to delete FINAL project reports.");
+
+            this.Context.ProjectReports.Remove(originalReport);
+            this.Context.CommitTransaction();
+        }
+
+
+        /// <summary>
+        /// Generate snapshots for all SPL projects using the passed 'to' date as the snapshot date.
+        /// </summary>
+        /// <param name="to"></param>
+        /// <returns></returns>
+        private IEnumerable<ProjectSnapshot> GenerateSnapshots(DateTime to)
```

Comment:
You don't need to pass `to` here.  We're simply generating a new collection of snapshots for the most current moment in time.

Reply:
This answer: https://github.com/bcgov/PIMS/pull/561/files/7986bf90439eb03cb11569b09887dcf430130f9b#r509670817
is also relevant for this question.